docs/dev/core/internal/payload.md

```diff
@@ -75,7 +75,45 @@ A [Timing distribution](../../../user/metrics/timing_distribution.md) is represe
 | Field name | Type | Description |
 |---|---|---|
 | `sum` | Integer | The sum of all recorded values. |
-| `values` | Map&lt;String, Integer&gt; | The values in each bucket. The key is the minimum value for the range of that bucket. Buckets with no values are not reported. |
+| `values` | Map&lt;String, Integer&gt; | The values in each bucket. The key is the minimum value for the range of that bucket. |
+
+A contiguous range of buckets is always sent, so that the server can aggregate and visualize distributions, without knowing anything about the specific bucketing function used.
+This range starts with the first bucket with a non-zero accumulation, and ends at one bucket beyond the last bucket with a non-zero accumulation (so that the upper bound on the last bucket is retained).
```

Comment:
One last thing: With the current mechanism we have 313 or so buckets (based on the maximum recordable time, 10 minutes). If that last bucket is non-empty, does it mean we add a 314th (empty) bucket in the serialization?

Reply:
That's a good question.  I think probably no, because the last bucket is basically an overflow bucket and basically "unbounded".  Will need to circle back on the implementation and make sure that's what it does...